File: app/src/main/java/com/ninhttd/devtest/presentation/product/adapter/ThongSoKiThuatAdapter.java
package com.ninhttd.devtest.presentation.product.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.ninhttd.devtest.R;
import com.ninhttd.devtest.presentation.product.model.AttributeGroup;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ThongSoKiThuatAdapter extends RecyclerView.Adapter<ThongSoKiThuatAdapter.ViewHolder> {

    List<AttributeGroup> data;
    FragmentActivity fragmentActivity;

    public ThongSoKiThuatAdapter(List<AttributeGroup> data, FragmentActivity fragmentActivity) {
        this.data = data;
        this.fragmentActivity = fragmentActivity;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_attribute_group_item, parent, false);
        return new ThongSoKiThuatAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        if (data.size() > 0) {
            if(position%2==0){
                holder.lnParent.setBackground(fragmentActivity.getResources().getDrawable(R.drawable.bg_rounded_pale_grey_8dp));
            }else{
                holder.lnParent.setBackground(fragmentActivity.getResources().getDrawable(R.drawable.bg_white));
            }
            AttributeGroup item = data.get(position);
            holder.txtTitle.setText(item.getName());
            holder.txtValue.setText(item.getValue());
        }
    }

    @Override
    public int getItemCount() {
        return data != null ? data.size() : 0;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.lnParent)
        LinearLayout lnParent;
        @BindView(R.id.txt_title)
        TextView txtTitle;
        @BindView(R.id.txt_value)
        TextView txtValue;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/ninhttd/devtest/presentation/product/adapter/ThongSoKiThuatAdapter.kt
package com.ninhttd.devtest.presentation.product.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView

import com.ninhttd.devtest.R
import com.ninhttd.devtest.data.entity.AttributeGroup

import butterknife.BindView
import butterknife.ButterKnife

class ThongSoKiThuatAdapter(var data: List<AttributeGroup>?, var fragmentActivity: FragmentActivity) : RecyclerView.Adapter<ThongSoKiThuatAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.view_attribute_group_item, parent, false)
        return ThongSoKiThuatAdapter.ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (data!!.isNotEmpty()) {
            if (position % 2 == 0) {
                holder.lnParent!!.background = fragmentActivity.resources.getDrawable(R.drawable.bg_rounded_pale_grey_8dp)
            } else {
                holder.lnParent!!.background = fragmentActivity.resources.getDrawable(R.drawable.bg_white)
            }
            val item = data!![position]
            holder.txtTitle!!.text = item.name
            holder.txtValue!!.text = item.value
        }
    }

    override fun getItemCount(): Int {
        return if (data != null) data!!.size else 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @BindView(R.id.lnParent)
        @JvmField
        var lnParent: LinearLayout? = null
        @BindView(R.id.txt_title)
        @JvmField
        var txtTitle: TextView? = null
        @BindView(R.id.txt_value)
        @JvmField
        var txtValue: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
